### PR TITLE
check return err of node

### LINKED ISSUE #474: Not check return err of list nodes
Forget to check return err of node list

https://github.com/cloudtty/cloudtty/blob/0ea12ebbf3481c77c68ad18d1f691445466090d4/pkg/controllers/cloudshell_controller.go#L459
```
	if len(nodes.Items) == 0 {
		if err := c.List(ctx, nodes, client.MatchingLabels{"node-role.kubernetes.io/control-plane": ""}); err != nil || len(nodes.Items) == 0 {
			return "", err
		}
	}

```

### PR DESCRIPTION
**Pull request by jxs1211**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #474

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/cloudshell_controller.go [hunk 1]
```diff
@@ -449,14 +449,16 @@ func (c *Controller) CreateRouteRule(ctx context.Context, cloudshell *cloudshell
...
-	// "node-role.kubernetes.io/cotrol-plane".
+	// "node-role.kubernetes.io/control-plane".
 	nodes := &corev1.NodeList{}
 	if err := c.List(ctx, nodes, client.MatchingLabels{"node-role.kubernetes.io/master": ""}); err != nil {
 		return "", err
 	}
 	if len(nodes.Items) == 0 {
 		if err := c.List(ctx, nodes, client.MatchingLabels{"node-role.kubernetes.io/control-plane": ""}); err != nil || len(nodes.Items) == 0 {
-			return "", err
+			if err != nil {
```
Comment (calvin0327) [inline pkg/controllers/cloudshell_controller.go:459]: Why do we need to determine if err is empty again?

Comment (jxs1211) [reply pkg/controllers/cloudshell_controller.go]: > Why do we need to determine if err is empty again?

I mistakenly missed the inline checking, the updated is remove unnecessary length checking.

Comment (flpanbin) [reply pkg/controllers/cloudshell_controller.go]: Logically, this code isn't necessary. But I think keeping this code might make the meaning clearer.

Comment (jxs1211) [reply pkg/controllers/cloudshell_controller.go]: I think it's not necessary, maybe it can be closed.

Comment (flpanbin) [reply pkg/controllers/cloudshell_controller.go]: other fixes are needed. You can update it if you feel like continuing.

**Pull request closed without merging**